examples/z_pub.c

```diff
@@ -83,7 +83,7 @@ int main(int argc, char** argv) {
         z_publisher_put_options_default(&options);
 
         z_owned_bytes_t payload;
-        z_bytes_copy_from_str(&payload, buf);
+        z_bytes_from_str(&payload, buf, NULL, NULL);
```

Comment:
this is not really encouraged as explained later. Given that buf string is not constant, this might have unexpected consequences in case of session-local communications (when zenoh can keep/use pointer to the data indefinitely) that will be a pain to debug.

Reply:
I see. Let's leave it as it is.

Reply:
Yes this is ok, I would likely provide an explanation here, that this will not make a copy and only keep pointer to static string because it assumed to outlive everything.